### ISSUE TITLE
App crash. iOS 11

### ISSUE THREAD

**Issue by alexanderkhitev** [xmartlabs/Eureka#1241]
Hello! I saw in the crashlytics app crash, which happened in the Eureka framework.

```
Fatal Exception: NSRangeException
0  CoreFoundation                 0x185f6bd38 __exceptionPreprocess
1  libobjc.A.dylib                0x185480528 objc_exception_throw
2  CoreFoundation                 0x185f04c44 _CFArgv
3  CoreFoundation                 0x185e34cc0 -[__NSArrayM removeObjectAtIndex:]
4  UIKit                          0x18f56238c -[_UITableViewUpdateSupport(Private) _setupAnimationsForDeletedCells]
5  UIKit                          0x18f55a238 -[_UITableViewUpdateSupport _setupAnimations]
6  UIKit                          0x18f559b90 -[UITableView _updateWithItems:updateSupport:]
7  UIKit                          0x18f5417e0 -[UITableView _endCellAnimationsWithContext:]
8  UIKit                          0x18f69c650 -[UITableView _updateSections:withUpdateAction:rowAnimation:headerFooterOnly:usingPresentationValues:]
9  UIKit                          0x18f53d7b8 -[UITableView reloadSections:withRowAnimation:]
10 Eureka                         0x10200d294 Section.reload(with : UITableViewRowAnimation) -> () (Section.swift:48)
11 itchable                       0x100d660a8 ProfileUserFormVC.(aboutRow(String) -> TextAreaRow).(closure #2) (ProfileUserFormVC.swift:485)
12 Eureka                         0x102001d74 RowType (RowType.swift:254)
13 Eureka                         0x102001df8 RowType
14 Eureka                         0x101fe89b8 Row.updateCell() -> () (Row.swift:162)
15 Eureka                         0x101f89074 specialized FormViewController.tableView(UITableView, cellForRowAt : IndexPath) -> UITableViewCell (Core.swift:884)
16 Eureka                         0x101f7ff28 @objc FormViewController.tableView(UITableView, cellForRowAt : IndexPath) -> UITableViewCell (Core.swift)
17 UIKit                          0x18f6b0474 -[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:]
18 UIKit                          0x18f6b09d8 -[UITableView _createPreparedCellForGlobalRow:willDisplay:]
19 UIKit                          0x18f690670 -[UITableView _updateVisibleCellsNow:isRecursive:]
20 UIKit                          0x18f44bf10 -[UITableView layoutSubviews]
21 UIKit                          0x18f378000 -[UIView(CALayerDelegate) layoutSublayersOfLayer:]
22 QuartzCore                     0x189f490b4 -[CALayer layoutSublayers]
23 QuartzCore                     0x189f4d194 CA::Layer::layout_if_needed(CA::Transaction*)
24 QuartzCore                     0x189ebbf24 CA::Context::commit_transaction(CA::Transaction*)
25 QuartzCore                     0x189ee2340 CA::Transaction::commit()
26 QuartzCore                     0x189e370d4 CA::Display::DisplayLink::dispatch_items(unsigned long long, unsigned long long, unsigned long long)
27 IOKit                          0x1861d630c IODispatchCalloutFromCFMessage
28 CoreFoundation                 0x185efa080 __CFMachPortPerform
29 CoreFoundation                 0x185f149dc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__
30 CoreFoundation                 0x185f140e0 __CFRunLoopDoSource1
31 CoreFoundation                 0x185f11bb4 __CFRunLoopRun
32 CoreFoundation                 0x185e322d8 CFRunLoopRunSpecific
33 GraphicsServices               0x187cc3f84 GSEventRunModal
34 UIKit                          0x18f3de880 UIApplicationMain
35 itchable                       0x100c29014 main (AppDelegate.swift:23)
36 libdyld.dylib                  0x18595656c start
```

![eureka0](https://user-images.githubusercontent.com/12511909/30794466-34eb6836-a1d9-11e7-9015-1cb6e3b33211.png)

Also using Xcode Organizer, I opened a problem place in the code in the project and Xcode points to this line.

```
extension Section {

    public func reload(with rowAnimation: UITableViewRowAnimation = .none) {
        guard let tableView = (form?.delegate as? FormViewController)?.tableView, let index = index else { return }
        tableView.reloadSections(IndexSet(integer: index), with: rowAnimation) // **There's a bug**
    }
}

```

![eureka 1](https://user-images.githubusercontent.com/12511909/30794615-507485fa-a1da-11e7-8127-f743a9aa1d57.png)

I failed for several attempts to play the app crash, I also want to inform you that it happened on iPhone 5s iOS 11.0.0 (15A372)

**Comment by mtnbarreto** [comment]
Not sure what's going wrong with the information provided. Were you able to reproduce the issue? Which swift, eureka version is the app using?

**Comment by alexanderkhitev** [comment]
Hi @mtnbarreto ! I use Swift 4 and Eureka 3.0.0. Look, I think this happens when the form already has sections and cells, and we add a cell to one of the sections.

**Comment by alexanderkhitev** [comment]
@mtnbarreto No, I have not been able to repeat it so I can write the result step-by-step, but if I can immediately write.

**Comment by mtnbarreto** [comment]
Eureka 4.0.0 works with swift 4. Are you sure you are using swift4?

**Comment by alexanderkhitev** [comment]
@mtnbarreto Yes, I sure 
![swift version](https://user-images.githubusercontent.com/12511909/31212262-92c273e4-a9b0-11e7-8e5e-fc43fe34c771.png)

**Comment by alexanderkhitev** [comment]
@mtnbarreto Sorry, I specified the wrong version of Eureka, since in Podfile.lock, why was 3.0.0 displayed instead of 4.0.0, I'm using version 4.0.0

**Comment by mtnbarreto** [comment]
Are you still having the trouble?  Please make sure Eureka is being compiled with swift 4. I just released a new version 4.0.1
